stdlib/http/src/main/ballerina/src/http/auth/authz_handler.bal

```diff
@@ -71,108 +72,26 @@ public type AuthzHandler object {
                 }
             }
 
-            var authorizedFromCache = self.authorizeFromCache(authzCacheKey);
+            var authorizedFromCache = auth:authorizeFromCache(authzCacheKey, self.positiveAuthzCache, self.negativeAuthzCache);
             if (authorizedFromCache is boolean) {
-                return authorizedFromCache;
+                authorized = authorizedFromCache;
             } else {
-                // if there are scopes set in the AuthenticationContext already from a previous authentication phase,
-                // try to match against those.
                 if (authCtxtScopes.length() > 0) {
-                    boolean authorized = checkForScopeMatch(scopes, authCtxtScopes, resourceName, method);
-                    // cache authz result
-                    self.cacheAuthzResult(authzCacheKey, authorized);
-                    return authorized;
+                    authorized = auth:checkForScopeMatch(scopes, authCtxtScopes);
```

Comment:
Shouldn't `checkForScopeMatch` take care-of first doing `authorizeFromCache` and at the end set the items to cache (`cacheAuthzResult`), without the HTTP module having to do that?

Reply:
Yes. That's better. Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/18042/commits/c07b2badcc6619064502c0d4e7b4711bc6faa948